rpcs3/Emu/Cell/SPUThread.cpp

```diff
@@ -556,6 +563,18 @@ void SPUThread::do_dma_transfer(const spu_mfc_cmd& args, bool from_mfc)
 	}
 	}
 
+		if (args.MSSync_type == 2) //MSSync from the channel requast.
+		{
+			MSSync_req_left--;
+			ch_event_stat |= (MSSync_req_left == 0 ? SPU_EVENT_MS : 0);
+		}
+		
+		else if (args.MSSync_type == 1)  //MSSync from spu register.
+		{
+			MSSync_reg_left--;
```

Comment:
Your increasing `MSSync_reg_left` for every command in the queue but only lowering it in this function (do_dma_transfer). Not every command hits this function does it? Also theres a race condition where you would flag and increase this variable, but the command is being processed somewhere after this line, but before its actually popped. This counter system will not work